AdaptorsDataAccessPlugins/test/unit/src/au/gov/asd/tac/constellations/dataaccess/adaptors/providers/gaffer/GafferSimpleQueryNGTest.java

```diff
@@ -0,0 +1,138 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellations.dataaccess.adaptors.providers.gaffer;
+
+import au.gov.asd.tac.constellation.graph.processing.GraphRecordStore;
+import au.gov.asd.tac.constellation.graph.processing.RecordStore;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import static org.testng.Assert.assertEquals;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
+import static uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser.DEFAULT_SERIALISER_CLASS_NAME;
+import uk.gov.gchq.gaffer.sketches.serialisation.json.SketchesJsonModules;
+
+import static org.mockito.Mockito.*;
+import org.openide.util.Exceptions;
+import uk.gov.gchq.gaffer.commonutil.iterable.CloseableIterable;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+
+/**
+ *
+ * @author GCHQDeveloper601
+ */
+public class GafferSimpleQueryNGTest {
+
+    public GafferSimpleQueryNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        JSONSerialiser.update(DEFAULT_SERIALISER_CLASS_NAME, SketchesJsonModules.class.getCanonicalName(), Boolean.TRUE);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Test of class ExtractFromContentPlugin.
+     *
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testAddResultsToRecordStore() {
+        GafferSimpleQuery simpleQuery = new GafferSimpleQuery();
+
+        final RecordStore recordStore = new GraphRecordStore();
+        List<Element> elements = fetchElementsFromFile("resources/exampleGafferResponseOneHop.json");
+        elements.forEach(e -> simpleQuery.addResultsToRecordStore(e, recordStore));
+        assertEquals(recordStore.size(), 3);
+    }
+    
+    @Test
+    public void testQueryForOneHop(){
+        final RecordStore recordStore = new GraphRecordStore();
+        GafferConnector connMock = mock(GafferConnector.class);
+        GafferSimpleQuery simpleQuery = new GafferSimpleQuery();
+        
+        List<String> queryIds = Arrays.asList("M4");
+        var opChain =simpleQuery.buildOneHopChain(queryIds);
```

Comment:
Probably less important for tests, but my previous comment about unclear types also applies here.